src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/ClientNotifierServiceBase.cs

```diff
@@ -3,24 +3,25 @@
 
 using System.Threading;
 using System.Threading.Tasks;
-using OmniSharp.Extensions.JsonRpc;
-using OmniSharp.Extensions.LanguageServer.Protocol.Models;
-using OmniSharp.Extensions.LanguageServer.Protocol.Server;
 
 namespace Microsoft.AspNetCore.Razor.LanguageServer
 {
     // The VSCode OmniSharp client starts the RazorServer before all of its handlers are registered
     // because of this we need to wait until everthing is initialized to make some client requests like
     // razor\serverReady. This class takes a TCS which will complete when everything is initialized
     // ensuring that no requests are sent before the client is ready.
-    internal abstract class ClientNotifierServiceBase: IOnLanguageServerStarted
+    internal abstract class ClientNotifierServiceBase : IOnLanguageServerStarted
     {
-        public abstract Task<IResponseRouterReturns> SendRequestAsync(string method);
+        public abstract Task<TResponse> SendRequestAsync<TParams, TResponse>(string method, TParams @params, CancellationToken cancellationToken);
 
-        public abstract Task<IResponseRouterReturns> SendRequestAsync<T>(string method, T @params);
+        public abstract Task SendNotificationAsync<TParams>(string method, TParams @params, CancellationToken cancellationToken);
+        public abstract Task SendNotificationAsync(string method, CancellationToken cancellationToken);
 
-        public abstract Task OnStarted(ILanguageServer server, CancellationToken cancellationToken);
+        public abstract Task OnStartedAsync(CancellationToken cancellationToken);
+    }
 
-        public abstract InitializeParams ClientSettings { get; }
+    internal interface IOnLanguageServerStarted
```

Comment:
When we start handling the endpoints we'll be calling each instance of `IOnLanguageServerStarted` (I'll probably change the name to `IOnInitialized` from the Initialized handler just like we did with Roslyn. This will let components that need to do one-time setup register through this stuff and guarantee only getting called once. More specifically, without this the `DefaultClientNotifierServiceBase` may try to start syncing documents to the Roslyn LS before we  have called `Initialized` which A) violates the spec B) doesn't work.